engine/classes/ElggEntity.php

```diff
@@ -1686,7 +1686,14 @@ protected function update() {
 		unset($persisted_entity);
 
 		if ($allow_edit) {
-			$allow_edit = _elgg_services()->events->trigger('update', $this->type, $this);
+			$allow_edit = elgg_trigger_before_event('update', $this->type, $this);
```

Comment:
```It will be a major pain point for plugin upgrades```
I wouldn't call it a major pain point - append ':before' or ':after' to the registration event.  Minor inconvenience at best :)

Having it's own entry in the logs is a decent point though.